### ISSUE TITLE
Scoreboard not being updated

### ISSUE THREAD

**Issue by mzegar** [TTTReborn/tttreborn#324]
## Prerequisites

- [X] I have searched [existing bug reports](https://github.com/TTTReborn/ttt-reborn/issues?q=label%3Aissue%2Fbug-report) and confirmed my issue is not already reported.

## Summary

No repo steps

```
theres also another bug where you can't see your traitor mate in scoreboard
doesn't tag him as red
and scoreboard still doesn't show everyone connected
```

**Comment by EmKaCe** [comment]
No players are being shown in the scoreboard when running with `net_fakelag 200` and `sv_lan 1`.
New players that join after you are not listed in the scoreboard at all.

**Comment by mzegar** [comment]
@EmKaCe Yeah thanks for this, we noticed the same thing happening with our inventory system. I'm assuming what happens is the data doesn't get transmitted to the client by the time we call the event to update the UI.

I currently have an issue open for the inventory slot issue.

**Comment by EmKaCe** [comment]
@mzegar From what I can tell so for (a bit of testing with logs and fakelag), it turns out that the client variable from [`ClientJoined(Client client)`](https://github.com/TTTReborn/tttreborn/blob/a02fa2db8dee87f1aedaf0d0d43cf57f89a17192/code/gamemode/Game.cs#L111) doesn't get passed to [`OnPlayerConnected(Client client)`](https://github.com/TTTReborn/tttreborn/blob/caf20f43f82594ce59d4663338be27450079a5af/code/ui/generalhud/scoreboard/Scoreboard.cs#L96) reliably.
![image](https://user-images.githubusercontent.com/20030146/143720724-94863154-bd27-437d-abd3-6c7aeef66a12.png)

**Comment by mzegar** [comment]
@EmKaCe sounds like it's basically this issue then https://github.com/Facepunch/sbox-issues/issues/1324

Thanks for looking into it

**Comment by mzegar** [comment]
Closing any issues since I am not a part of the project, please feel free to reopen any active issues.
